src/test/java/updated/game/abstractClasses/Board.java

```diff
@@ -0,0 +1,150 @@
+    }
+  
+    public void userMove(Gamer gamer, int row, int col){
+        gamer.incrementNumberOfMoves();
+        this.board.get(row).set(col, gamer.getSymbol());
+    }
+
+    public boolean checkDiagonal() {
+//        boolean primaryDiagonalResult = true;
+//        boolean secondaryDiagonalResult = true;
+//        char primaryDiagonalSymbol = this.board.getFirst().getFirst();
+//        char secondaryDiagonalSymbol = this.board.getFirst().getLast();
+//
+//        for(int row = 0; row < BOARD_SIZE; row++){
+//            for(int col = 0; col < BOARD_SIZE; col++){
+//                if(row == col){
+//                    primaryDiagonalResult = primaryDiagonalResult & (primaryDiagonalSymbol == this.board.get(row).get(col) && this.board.get(row).get(col) != ' ');
+//                    System.out.println(row + " " + col + " " + primaryDiagonalSymbol + " "+ primaryDiagonalResult);
+//
+//                }
+//                else if (col == (BOARD_SIZE-1-row)){
+//                    secondaryDiagonalResult = secondaryDiagonalResult & (secondaryDiagonalSymbol == this.board.get(row).get(col) && this.board.get(row).get(col) != ' ');
+//                    System.out.println(row + " " + col +" " + secondaryDiagonalSymbol + " " + secondaryDiagonalResult);
+//                }
+//            }
+//            if(primaryDiagonalResult){
+//                this.winningSym = primaryDiagonalSymbol;
+//                System.out.println("Primary Result: "+primaryDiagonalSymbol);
+//                break;
+//            }
+//            if(secondaryDiagonalResult){
+//                System.out.println("Secondary Result: "+secondaryDiagonalSymbol);
+//                this.winningSym = secondaryDiagonalSymbol;
+//                break;
+//            }
+//        }
+//
+//        return primaryDiagonalResult | secondaryDiagonalResult;
+        boolean result = board.getFirst().getFirst() == board.get(1).get(1) && board.get(1).get(1) == board.get(2).get(2);
+        result = result || board.get(0).get(2) == board.get(1).get(1) && board.get(1).get(1) == board.get(2).getFirst();
+        if (result) {
+            winningSym = board.get(1).get(1);
+        }
+        return result && winningSym != ' ';
+    }
+
+    public boolean checkVertical() {
+        boolean result = true;
+        char currentSymbol = ' ';
+        for(int col = 0; col < BOARD_SIZE; col++){
+            currentSymbol = this.board.getFirst().get(col);
+            result = true;
+            for(int row = 0; row < BOARD_SIZE; row++){
+                result = result & (this.board.get(row).get(col) == currentSymbol && this.board.get(row).get(col) != ' ');
+            }
+            if(result){
+                System.out.println("In Check vertical: " + currentSymbol);
+                this.winningSym = currentSymbol;
+                break;
+            }
+        }
+        return this.winningSym != ' ';
+    }
+
+    public boolean checkHorizontal() {
```

Comment:
Horizontal and vertical checks have similar logic. You can reduce redundancy by creating a common method and applying it in both cases.